### PR TITLE
Enable `-field` in `vault kv get/put`

### LINKED ISSUE #4424: -field option not supported by 'vault kv get'
Using `-field` flag from `vault read` is quite useful to avoid calling `jq` when using KV secrets engine V1. It is not possible to use `-field` or similar option for KV secrets engine V2 (ie. `vault kv get`).

**Environment:**
* Vault Version: Vault v0.10.0 ('5dd7f25f5c4b541f2da62d70075b6f82771a650d')
* Operating System/Architecture: Debian GNU/Linux 8 (jessie) / x86_64

**Expected Behavior:**
Vault CLI returns a specific field on `vault kv get --field some/path`.

**Actual Behavior:**
Vault CLI returns following error message `flag provided but not defined: -field`.

**Steps to Reproduce:**
`vault kv get -field=world secret/hello`

**References:**
[Vault documentation](https://www.vaultproject.io/intro/getting-started/first-secret.html#reading-a-secret) suggest it is possible to do so.

### PR DESCRIPTION
**Pull request by jefferai**
Fixes #4424

### CODE REVIEW

File: command/util.go [hunk 1]
```diff
@@ -76,11 +76,18 @@ func RawField(secret *api.Secret, field string) (interface{}, bool) {
...
+	switch data.(type) {
+	case *api.Secret:
+		var ok bool
+		val, ok = RawField(data.(*api.Secret), field)
+		if !ok {
+			ui.Error(fmt.Sprintf("Field %q not present in secret", field))
+			return 1
+		}
+	case map[string]interface{}:
+		val = data.(map[string]interface{})[field]
```
Comment (calvn) [inline command/util.go]: Missing an ok check in here?

Comment (jefferai) [reply command/util.go]: It's a type switch :-)

Comment (calvn) [reply command/util.go]: I mean wrt accessing the [field] value in the data map. If the provided key doesn't exists, val will be the zero value (not sure what that would for interface{}).

Comment (jefferai) [reply command/util.go]: I think I see what you meant, not the conversion but the field lookup. Check the follow-up commit.

Comment (calvn) [reply command/util.go]: Nice, that's clean.

**Comment by queglay** [comment]
I was thinking because of this ticket it wouldbe possible to replace a single field at a secret path, but when I attempt to use vault kv put with -field it doesn't seem to work in vault 1.5.5

```
admin:((e4c08a0...)) $ vault kv get developers/dev/config/deadline_version
====== Metadata ======
Key              Value
---              -----
created_time     2020-11-21T23:27:24.159648302Z
deletion_time    n/a
destroyed        false
version          9

======= Data =======
Key            Value
---            -----
default        10.1.9.0
description    The version of the deadline installer.
example_1      10.1.9.2
value          10.1.9.0

admin:((e4c08a0...)) $ vault kv put -field=value developers/dev/config/deadline_version 10.1.9.5
Failed to parse K=V data: invalid key/value pair "10.1.9.5": format must be key=value

admin: ((e4c08a0...)) $ vault kv put -field developers/dev/config/deadline_version value=10.1.9.4
Must supply data

admin:((e4c08a0...)) $ vault kv put -field=value developers/dev/config/deadline_version value=10.1.9.4                                                                               
Field "value" not present in secret

admin:((e4c08a0...)) $ vault kv get developers/dev/config/deadline_version
====== Metadata ======
Key              Value
---              -----
created_time     2020-11-21T23:36:38.996604909Z
deletion_time    n/a
destroyed        false
version          10

==== Data ====
Key      Value
---      -----
value    10.1.9.4
```

**Pull request merged**